### DISCUSSION TITLE
Any way to make Driver.open / Driver.get / Driver.uc_open faster?

### DISCUSSION THREAD

**Discussion by oculunto** [seleniumbase/SeleniumBase#2157] [Q&A] [answered]
Hello, 
 I am using seleniumBase with the Driver object from seleniumbase import Driver 
 My problem (little inconvenience really...) is that after calling Driver.open / Driver.get / Driver.uc_open there is a few seconds between that and the next action. 
 Let me try to explain with an example: 
 
 ```python
 from seleniumbase import Driver
 browser = Driver(uc=True,user_data_dir = profileDir, proxy=PROXY_USER+":"+PROXY_PASS+"@"+PROXY_HOST+":"+PROXY_PORT)
 time.sleep(2)
 Driver.open(someURL)
 time.sleep(0.5)
 #Here is where the "problematic" waiting time happens before the click
 elementToClick = browser.find_elements(By.XPATH, "//*[contains(text(),'something')]")
 if elementToClick != []:
      elementToClick[0].click()
 ```
 Keep in mind this is just a quick example, I usually use WebDriverWait(browser, 8).until(EC.visibility_of_element_located((By.CSS_SELECTOR, 'something'))) to speed things up.
 I have tried with many of the methods from Driver (open, get, uc_open) but it seems to have a few seconds delay after opening the page that I would like to reduce for my use case if possible. 
 I also tried using "none" as page_load_strategy to no avail. 
 I tried to look in the code if the get method is redefined and adds things that might induce delay, but as a beginner i wasn't able to find what.
 If anyone knows how I can modify any of those methods to reduce the delay (probably at the cost of some stability), it'd be cool.
 I appreciate the help and the work of the author !

**Comment by mdmintz** [comment, marked as answer]
One of the things that SeleniumBase does to avoid detection is to:

* Disconnect chromedriver temporarily from Chrome before loading websites that try to detect chromedriver.

(It's one of the things listed in https://github.com/seleniumbase/SeleniumBase/issues/2000)

It's only done when a `requests.get()` call returns a `403`, or similar. If that happens, chromedriver disconnects from the browser as the web page gets loaded, and then reconnects as quickly as possible so that selenium can resume performing actions. The default `driver.get()` method is modified with this change.

To perform the original `driver.get()` call (which doesn't have this extra wait), use `self.driver.default_get()` instead.

To summarize, use `self.driver.default_get(URL)` when in UC Mode to prevent the special disconnect. Opening URLs will go faster, but websites may be able to detect that you're using chromedriver.
